### PR TITLE
Code abstraction

### LINKED ISSUE #33: Code Abstraction
- Segregrate the code based on <tag>, <title>, <author>,<readtime>,<data>,<photo>,<content>
code source:
/blog
<id>.js like (1 js, 2 js, 3js)

### PR DESCRIPTION
**Pull request by amitg01**
fixes #33

### CODE REVIEW

**Comment by n1c01a5** [comment]
@amitg01 change the versioning in package.json: `0.1` => `0.0.1` to fix the vercel build

File: pages/blog/[slug].js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import BlogHeader from '../../components/elements/BlogHeader'
+import { getPosts } from '../api/posts'
+
+import RecoverBlog from '../api/posts/recover'
+import EscrowBlog from '../api/posts/escrow'
+
+const blogsMap = {
+  'lost-and-found-iphone': RecoverBlog,
+  'securing-valuables-with-escrow-smart-contracts': EscrowBlog
+}
```
Comment (n1c01a5) [inline pages/blog/[slug].js:12]: We don't need this mapping.
The HTML content has to be in the `../api/posts/index.js`.

But you can create a different file like you didi it and import in this file and add an attribute like this: `content: htmlContentEscrow`.

Comment (amitg01) [reply pages/blog/[slug].js:12]: @bhavesh92pandey @n1c01a5 Next.js doesn't allow JSX elements to be exported inside an object. So importing `htmlContentEscrow` and using it inside a `post` object in the `../api/posts/index.js` like `content: htmlContentEscrow` is not working. I'll try to figure out a non-complex way of doing it. I'll keep you posted about the progress.

Comment (n1c01a5) [reply pages/blog/[slug].js:12]: @amitg01 I think you can insert html with `dangerouslySetInnerHTML`.

Comment (amitg01) [reply pages/blog/[slug].js:12]: @n1c01a5 The problem here is with the storage of the content and not the usage. Like I mentioned earlier, Next.js does not allow something like `content: htmlContentEscrow` in an individual post object in `../api/posts/index.js`, here `htmlContentEscrow` is a jsx element. 

If it was allowed, I could have used `dangerouslySetInnerHTML` and set the content after the post object had been loaded.

We can schedule a call to discuss this if there is some doubt.

Comment (n1c01a5) [reply pages/blog/[slug].js:12]: Ok I will do some tests and I will answer you.

Comment (amitg01) [reply pages/blog/[slug].js:12]: Cool.

**Comment by n1c01a5** [comment]
@amitg01 can you check my code. The html must be a string. It's normal because it will be store in a database later.
My code is  just a POC, clean it and add the rest of feature like the cover for example.

Take care to all the code has to be abstract.

File: pages/blog/[slug].js (hunk 1 above)
Comment (n1c01a5) [reply pages/blog/[slug].js]: See my code under the comment.

**Comment by amitg01** [comment]
@n1c01a5 Can I use SCSS to clean up the CSS code as well? It has now become over 500 lines long and is getting more complicated to maintain. I'll have to install the `node-sass` package for this.

**Comment by n1c01a5** [comment]
Ok let's do it!

Le ven. 1 janv. 2021 à 19:49, Amit Kumar Gupta <notifications@github.com> a
écrit :

**Comment by n1c01a5** [comment]
@amitg01 on the other hand we already have a preprocessor (styled-jsx) so we must make sure that there will be no conflict between the preprocessors and also have consistent code.

**Comment by n1c01a5** [comment]
@amitg01 For `sass` it will better to create a new branch `refactor(style): move to sass` to have small pull request. It easier for the review and for debugging if needed.

**Comment by bhavesh92pandey** [comment]
https://github.com/n1c01a5/recover-website/issues/49

**Comment by amitg01** [comment]
@n1c01a5 Please review.

Due to the changes, there are some minor issues with the UI.
Screenshot 2021-01-04 at 6 42 37 PM

As you can see here, the padding is a little off. But to fix issues like these, a lot of css has to be changed. And since we are going to migrate to SCSS, these changes are redundant. These minor changes can be taken care of when the css is completely refactored. Please let me know what you think.

**Comment by bhavesh92pandey** [comment]
@amitg01 

Yes, you can go forward with SCSS deployment so we can see the final clean code. This is okay for now

**Comment by n1c01a5** [comment]
@amitg01  yes I agree with @bhavesh92pandey .

**Comment by n1c01a5** [comment]
@amitg01 For the migration `css` to `scss` can you create a new PR. Thanks

**Pull request merged**